services/blockstorage/sync/state_idle_test.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 func TestIdleStateStaysIdleOnCommit(t *testing.T) {
-	h := newBlockSyncHarness()
+	h := newBlockSyncHarness().withNoCommitTimeout(time.Second) // we are checking for a newly created state, the timeout here is irrelevant
```

Comment:
why does `withNoCommitTimeout` take a `Duration`?

Reply:
it is the timeout interval for when the sync is supposed to transition to the next state
default for tests is 3ms i think
i added this to solve a flaky case where the next state function will deprioritize and then it would take more than 3ms to complete, and the state would change